larray_editor/arraywidget.py

```diff
@@ -698,9 +698,13 @@ def set_filters(self):
             filters_layout.addStretch()
         self.data_adapter.update_filtered_data({})
 
-    def _update(self):
+    def _update_digits_scientific(self, data):
+        """
+        data : LArray
+        """
         # TODO: Adapter must provide a method to return a data sample as a Numpy array
-        data = self.data_adapter.get_data().data
+        assert isinstance(data, la.LArray)
+        data = data.data
```

Comment:
OK for now but my objective is that, one day, the ArrayEditorWidget can be used to display not only LArray objects... 
In other words, ArrayEditorWidget should only manipulate an instance of the Adapter and never deal with LArray objects (but once again, this is not medium/long term objective)

Reply:
I know, and this is my wish too, but currently there you had an implicit assumption that the data was an LArray (by using the .data). I made it explicit, which is more readable and less error-prone.